core/src/test/scala/unit/kafka/server/MetadataRequestTest.scala

```diff
@@ -234,6 +235,32 @@ class MetadataRequestTest extends AbstractMetadataRequestTest {
     }
   }
 
+  @Test
+  def testInvalidMetadataRequestReturnsError(): Unit = {
```

Comment:
Do we need a full blown slow request test for this or can we rely on unit tests only?

Reply:
This was one way to test the KafkaApis code, but I suppose I could move this to a unit test that only tests the method itself (and not the whole request path)

Reply:
I guess the only other reason I tested the whole path was to make sure the response could be sent back (if the name was null, it could not have), but it should suffice to also have a non-null check.

Reply:
if you think it's valuable, we can keep it. But can we piggy back on another test that tests invalid things? Then we'd save starting and stopping a kafka cluster.

Reply:
I must have missed this comment earlier. I moved the test to KafkaApisTest. There we don't check if the response is built since that is part of the request/response handling code. From what I saw the other invalid tests were very focused on one thing. 

I think the biggest question is whether we care to test the null name is correctly being set to the empty string. I agree it is good not to start and stop another kafka cluster.